### ISSUE TITLE
Workspace `summary` method can fail when some samples have no gates

### ISSUE THREAD

**Issue by milesdai** [whitews/FlowKit#155]
**Describe the bug**
If a workspace has some samples that do not have gates, then the `Workspace.summary()` method can fail with a bad dictionary lookup. This looks like it might be follow up to #91.

During the wsp import, any samples that do not have gates are excluded from the returned sample dictionary:
https://github.com/whitews/FlowKit/blob/e539236c7b480582b80eeacbf47533ff12bb19de/flowkit/_utils/wsp_utils.py#L600-L604

However, this sample may still be present in the groups (e.g. `All Samples`). Later on, when `Workspace.summary()` is called, the first sample in the group is used as an example to get the gate depth:

https://github.com/whitews/FlowKit/blob/e539236c7b480582b80eeacbf47533ff12bb19de/flowkit/_models/workspace.py#L124-L133

If the sample is excluded from sample_data, then `get_gating_strategy` will fail the dictionary lookup.

I'm not familiar enough with the library yet to know the best way forward, but it seems like maybe the solution might be either:
1. Add the sample to the sample dictionary with an empty `GatingStrategy` if it has no gates instead of skipping it
2. Add logic to the summary method to handle missing samples.

**Code To Reproduce**
Code to reproduce the behavior:
```python
import flowkit as fk
wsp_path = 'path/to/wsp_file'
sample_path = 'path/to/fcs/directory'
wsp = fk.Workspace(wsp_path, fcs_samples = sample_path)
print(wsp.summary())
```

**Expected behavior**
A summary should be printed.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: macOS
 - Python version: 3.9.13
 - FlowKit version: 1.0.1

**Additional context**
Traceback  (some paths redacted):
```
Traceback (most recent call last):
  File "<...>/main.py", line 28, in <module>
    print(wsp.summary())
  File "<...>/flowkit/_models/workspace.py", line 132, in summary
    gs = self.get_gating_strategy(example_sample_id)
  File "<...>/flowkit/_models/workspace.py", line 285, in get_gating_strategy
    return copy.deepcopy(self._sample_data_lut[sample_id]['gating_strategy'])
KeyError: '<filename>.fcs'
```

**Comment by whitews** [comment]
Hi Miles,

Thanks for reporting the bug and for digging in a bit to see what is going on. I'll have to make a Workspace in FlowJo to test out the best fix. So I make an equivalent scenario, is it correct that you have a sample in the 'All Samples' group that doesn't belong to any other group and doesn't have any gates?

Thanks,
Scott

**Comment by milesdai** [comment]
Yup I believe that's correct!

**Comment by whitews** [comment]
Hey Miles,

I've fixed this in the latest develop branch. It was a bit tricky to replicate because the sample without gates has to be the first one parsed. The fix was implemented by allowing a "blank" GatingStrategy to be created for the gate-less sample. This has the benefit of allowing the retrieval of the comp matrix and transforms assigned to the sample within the workspace file.

However, this behavior of using the 1st sample is a admittedly janky and raises the question of whether the format for the returned summary should be changed. The current format is group based and was lifted from the Session class where variations on the group gate structure is not allowed. Given that the sample gate structure can vary within a workspace group I'm wondering if it would be better to make the output sample based, i.e. the rows are per sample rather than per group. The same columns for gate count & level could remain, with maybe the addition of a group column that would be a list of groups each sample belongs to. It might not look as nice but would be useful for seeing if the gate counts differed.
